sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala

```diff
@@ -311,6 +317,29 @@ class HadoopTableReader(
     // Only take the value (skip the key) because Hive works only with values.
     rdd.map(_._2)
   }
+
+  /**
+   * Creates a NewHadoopRDD based on the broadcasted HiveConf and other job properties that will be
+   * applied locally on each slave.
+   */
+  private def createNewHadoopRdd(tableDesc: TableDesc, path: String): RDD[Writable] = {
```

Comment:
I think these two new methods could just be inlined into createHadoodRdd to reduce a little duplication.

Reply:
It seems no duplication in these two methods as I tried moving them into createHadoodRdd.

Reply:
NewHadoopRDD needs a JobConf and HadoopRDD needs a function(initializeJobConfFunc). The inputFormatClass of these two methods are also different.

Reply:
The two new methods have the same arguments and call tableDesc.getInputFileFormatClass and rdd.map(_._2); not a big deal, more concerned with keeping code locality. I don't feel strongly about it. How about importing these Hadoop classes with a rename, rather than use the fully qualified names several times?

Reply:
Yes, renamed these classes and changed `java.lang.Class` to `Class`.